anvill/python/anvill/util.py

```diff
@@ -14,26 +14,67 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 
-_DEBUG_FILE = None
-_DEBUG_PREFIX = ""
+import os
+import sys
+import logging
 
 
-def INIT_DEBUG_FILE(file):
-    global _DEBUG_FILE
-    _DEBUG_FILE = file
+# Create a logger
+logger = logging.getLogger(__name__)
```

Comment:
Does use of `__name__` here always make it so that the logger has the name `anvill.util`? I am not 100% sure, but I don't think name propagates from higher-level modules.

Reply:
So typically you want the logger name to be a full module name based on the calling module, like "anvill.binja.bnprogram", "anvill.type", etc. so that you can filter logs by module name, and like set a logging config to only log certain modules, or only make some log parts verbose and not all. 

If __name__ is indeed always anvill.util, would it be possible to make a function in here, say, something like:

```python
def createLogger(logger_name):
  logger = logging.getLogger(logger_name)
  #... do setup/config here as needed ...
  return logger.info, logger.warning, logger.error, logger.fatal

#Then at the start of each module, you'd have

from .util import createLogger

INFO, WARN, DEBUG, FATAL = createLogger(__name__)

```

Reply:
@artemdinaburg, Thanks for the review. You are correct. The logger name here is `anvil.util`. It creates a single logger for all the modules. 

If you would like to have a separate logger for each module, it's a good idea to go ahead with the logger configuration file. I will have it in place and have the `root` logger as default.

Reply:
I update the design and hopefully, it should work as you described. The logger can be configured with the `logging.ini` config file. If the file is missing, it falls back to the basic configuration.

There is an interface to create a module-level logger with the default configuration and It can be configured with `logging.ini` file. Please let me know your thoughts.